sdk/resourcemanager/azure-resourcemanager-cdn/src/main/java/com/azure/resourcemanager/cdn/implementation/CdnEndpointImpl.java

```diff
@@ -166,17 +169,22 @@ public Mono<CdnEndpoint> updateResourceAsync() {
             .updateAsync(this.parent().resourceGroupName(), this.parent().name(), this.name(),
                 endpointUpdateParameters);
 
-        Flux<CustomDomainInner> customDomainCreateTask = Flux.fromIterable(this.customDomainList)
+        Flux<CustomDomainInner> customDomainCreateTask = Flux.fromIterable(this.customDomainHostnames)
             .flatMapDelayError(itemToCreate -> this.parent()
                 .manager()
                 .serviceClient()
                 .getCustomDomains()
                 .createAsync(this.parent().resourceGroupName(), this.parent().name(), this.name(),
                     self.parent().manager().resourceManager().internalContext().randomResourceName("CustomDomain", 50),
-                    new CustomDomainParameters().withHostname(itemToCreate.hostname())),
+                    new CustomDomainParameters().withHostname(itemToCreate)),
                 32, 32);
 
-        Flux<CustomDomainInner> customDomainDeleteTask = Flux.fromIterable(this.deletedCustomDomainList)
+        Flux<CustomDomainInner> customDomainDeleteTask = this.parent()
+            .manager()
+            .serviceClient()
+            .getCustomDomains()
+            .listByEndpointAsync(this.parent().resourceGroupName(), this.parent().name(), this.name())
+            .filter(customDomain -> this.deletedCustomDomainHostnames.contains(customDomain.hostname()))
             .flatMapDelayError(itemToDelete -> this.parent()
                 .manager()
                 .serviceClient()
```

Comment:
Should it under condition that `deletedCustomDomainHostnames` is not empty?

Reply:
Yes, we should. It'll save a list call in update if no custom domains to delete.

Reply:
updated